src/test/java/mate/academy/dao/RoleDaoTest.java

```diff
@@ -0,0 +1,75 @@
+package mate.academy.dao;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Optional;
+import mate.academy.dao.impl.RoleDaoImpl;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Role;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RoleDaoTest extends AbstractTest {
+    private static final String USER = "USER";
+    private static final String NOT_USER = "NOT_USER";
+    private RoleDao roleDao;
+    private Role userRole;
+    private Role actual;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[] {Role.class};
+    }
+
+    @BeforeEach
+    void setUp() {
+        roleDao = new RoleDaoImpl(getSessionFactory());
```

Comment:
Creating new dao instance for each test is not required and will affect performance. 1 instance for class is enough

Reply:
not fixed

Reply:
Can you help me with th it? Because I can't do it, creating dao object require calling getSessionFactory() and this method is not static, so I can not move it to beforeAll. And I don't know any other ways to create dao object once per class, not before every test